src/test/script_standard_tests.cpp

```diff
@@ -0,0 +1,746 @@
+    expected.clear();
+    result = GetScriptForDestination(CNoDestination());
+    BOOST_CHECK(result == expected);
+
+    // GetScriptForRawPubKey
+    expected.clear();
+    expected << ToByteVector(pubkeys[0]) << OP_CHECKSIG;
+    result = GetScriptForRawPubKey(pubkeys[0]);
+    BOOST_CHECK(result == expected);
+
+    // GetScriptForMultisig
+    expected.clear();
+    expected << OP_2 <<
+        ToByteVector(pubkeys[0]) <<
+        ToByteVector(pubkeys[1]) <<
+        ToByteVector(pubkeys[2]) <<
+        OP_3 << OP_CHECKMULTISIG;
+    result = GetScriptForMultisig(2, std::vector<CPubKey>(pubkeys, pubkeys + 3));
+    BOOST_CHECK(result == expected);
+
+    // GetScriptForWitness
+    CScript witnessScript;
+
+    witnessScript << ToByteVector(pubkeys[0]) << OP_CHECKSIG;
+    expected.clear();
+    expected << OP_0 << ToByteVector(pubkeys[0].GetID());
+    result = GetScriptForWitness(witnessScript);
+    BOOST_CHECK(result == expected);
+
+    witnessScript.clear();
+    witnessScript << OP_DUP << OP_HASH160 << ToByteVector(pubkeys[0].GetID()) << OP_EQUALVERIFY << OP_CHECKSIG;
+    result = GetScriptForWitness(witnessScript);
+    BOOST_CHECK(result == expected);
+
+    witnessScript.clear();
+    witnessScript << OP_1 << ToByteVector(pubkeys[0]) << OP_1 << OP_CHECKMULTISIG;
+
+    uint256 scriptHash;
+    CSHA256().Write(&witnessScript[0], witnessScript.size())
+        .Finalize(scriptHash.begin());
+
+    expected.clear();
+    expected << OP_0 << ToByteVector(scriptHash);
+    result = GetScriptForWitness(witnessScript);
+    BOOST_CHECK(result == expected);
+}
+
+BOOST_AUTO_TEST_CASE(script_standard_IsMine)
+{
+    CKey keys[2];
+    CPubKey pubkeys[2];
+    for (int i = 0; i < 2; i++) {
+        keys[i].MakeNewKey(true);
+        pubkeys[i] = keys[i].GetPubKey();
+    }
+
+    CPubKey uncompressedPubkey0(pubkeys[0]);
+    uncompressedPubkey0.Decompress();
+
+    CScript scriptPubKey;
+    isminetype result;
+    bool isInvalid;
+
+    // P2PK compressed
+    {
+        CBasicKeyStore keystore;
+        scriptPubKey.clear();
+        scriptPubKey << ToByteVector(pubkeys[0]) << OP_CHECKSIG;
+
+        // Keystore does not have key
+        result = IsMine(keystore, scriptPubKey, isInvalid);
+        BOOST_CHECK_EQUAL(result, ISMINE_NO);
+        BOOST_CHECK(!isInvalid);
+
+        // Keystore has key
+        keystore.AddKey(keys[0]);
+        result = IsMine(keystore, scriptPubKey, isInvalid);
+        BOOST_CHECK_EQUAL(result, ISMINE_SPENDABLE);
+        BOOST_CHECK(!isInvalid);
+    }
+
+    // P2PK uncompressed
+    {
+        CBasicKeyStore keystore;
+        scriptPubKey.clear();
+        scriptPubKey << ToByteVector(uncompressedPubkey0) << OP_CHECKSIG;
+
+        // Keystore does not have key
+        result = IsMine(keystore, scriptPubKey, isInvalid);
+        BOOST_CHECK_EQUAL(result, ISMINE_NO);
+        BOOST_CHECK(!isInvalid);
+
+        // Keystore has key
+        keystore.AddKeyPubKey(keys[0], uncompressedPubkey0);
```

Comment:
You're adding a compressed CKey (`keys[0]`) with an uncompressed CPubKey, which is incorrect use of the function. You should first change the CKey's compressedness.

Perhaps this warrants a comment on AddKeyPubKey or even an assert. It should alleviate the need for a6564cca7d443789671778f769a24bbb5a4a0652 though.

Reply:
Thanks, will change.